hello-vulkan/app/src/main/cpp/vk_triangle/hellovk.h

```diff
@@ -0,0 +1,1279 @@
+  extensions.push_back("VK_KHR_android_surface");
+  if (enableValidationLayers) {
+    extensions.push_back(VK_EXT_DEBUG_UTILS_EXTENSION_NAME);
+  }
+  return extensions;
+}
+
+void HelloVK::createInstance() {
+  if (enableValidationLayers && !checkValidationLayerSupport()) {
+    throw std::runtime_error("validation layers requested, but not available!");
+  }
+  auto requiredExtensions = getRequiredExtensions(enableValidationLayers);
+
+  VkApplicationInfo appInfo{};
+  appInfo.sType = VK_STRUCTURE_TYPE_APPLICATION_INFO;
+  appInfo.pApplicationName = "Hello Triangle";
+  appInfo.applicationVersion = VK_MAKE_VERSION(1, 0, 0);
+  appInfo.pEngineName = "No Engine";
+  appInfo.engineVersion = VK_MAKE_VERSION(1, 0, 0);
+  appInfo.apiVersion = VK_API_VERSION_1_0;
+
+  VkInstanceCreateInfo createInfo{};
+  createInfo.sType = VK_STRUCTURE_TYPE_INSTANCE_CREATE_INFO;
+  createInfo.pApplicationInfo = &appInfo;
+  //   createInfo.flags |= VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR;
+  createInfo.enabledExtensionCount = (uint32_t)requiredExtensions.size();
+  createInfo.ppEnabledExtensionNames = requiredExtensions.data();
+  createInfo.pApplicationInfo = &appInfo;
+
+  if (enableValidationLayers) {
+    VkDebugUtilsMessengerCreateInfoEXT debugCreateInfo{};
+    createInfo.enabledLayerCount =
+        static_cast<uint32_t>(validationLayers.size());
+    createInfo.ppEnabledLayerNames = validationLayers.data();
+    populateDebugMessengerCreateInfo(debugCreateInfo);
+    createInfo.pNext = (VkDebugUtilsMessengerCreateInfoEXT *)&debugCreateInfo;
+  } else {
+    createInfo.enabledLayerCount = 0;
+    createInfo.pNext = nullptr;
+  }
+  VkResult result = vkCreateInstance(&createInfo, nullptr, &instance);
+  if (result != VK_SUCCESS) {
+    throw std::runtime_error("failed to create instance with err " +
+                             std::to_string(result) + " !");
+  }
+
+  uint32_t extensionCount = 0;
+  vkEnumerateInstanceExtensionProperties(nullptr, &extensionCount, nullptr);
+  std::vector<VkExtensionProperties> extensions(extensionCount);
+  vkEnumerateInstanceExtensionProperties(nullptr, &extensionCount,
+                                         extensions.data());
+  LOGI("available extensions");
+  for (const auto &extension : extensions) {
+    LOGI("\t %s", extension.extensionName);
+  }
+};
+
+void HelloVK::createSurface() {
+  const VkAndroidSurfaceCreateInfoKHR create_info{
+      .sType = VK_STRUCTURE_TYPE_ANDROID_SURFACE_CREATE_INFO_KHR,
+      .pNext = nullptr,
+      .flags = 0,
+      .window = window.get()};
+
+  VkResult result = vkCreateAndroidSurfaceKHR(
+      instance, &create_info, nullptr /* pAllocator */, &surface);
+  if (result != VK_SUCCESS) {
+    throw std::runtime_error("failed to create surface with err " +
+                             std::to_string(result) + " !");
+  }
+}
+
+// BEGIN DEVICE SUITABILITY
+// Might be overkill for Android currently since they generally only have one
```

Comment:
Something like 
'// Functions to find a suitable physical device with to execute Vulkan commands.'
would be better